paddle/fluid/platform/device_context.cc

```diff
@@ -459,12 +459,13 @@ void MKLDNNDeviceContext::SetBlob(const std::string& name,
   auto key_it = sBlob->find(cur_input_shape_str);
 
   if (key_it == sBlob->end()) {
-    // In cache clearing mode, cur_input_shape_cache_size defines max pblob
-    // capacity
-    if ((tid == kMKLDNNSessionID_CacheClearing) &&
-        (sBlob->size() == static_cast<size_t>(cur_input_shape_cache_size))) {
-      VLOG(2) << "tid=" << tid
-              << ", remove all head blob of shape: " << sBlob->begin()->first;
+    // In cache clearing mode, cur_input_shape_cache_capacity defines
+    // max pblob capacity
+    if ((sid == kMKLDNNSessionID_CacheClearing) &&
+        (sBlob->size() ==
```

Comment:
do you think">=" is better  here ?

Reply:
For batchsize=1, when `==`, it will clean. Thus `>` does not work. 
But for batchsize >1, I think `>=` is better. I will change it.

Reply:
when `==`, it will clean. Thus `>` does not work.
What's the case when `>` works?

Reply:
so far I still not see cases for >, but from code level, it looks safer to use >=.

Reply:
Got it.